### PR TITLE
Replace ThreadLocal with HashMap to eliminate server crash

### PR DESCRIPTION
**Pull request by ReinWD**
Workaround #1668 and #1722.

### CODE REVIEW

**Comment by me4502** [comment]
It's worth noting for the servers this affects, if worldedit doesn't have the threadlocal the server still dies from Brigadier's threadlocal

**Comment by phoenixlzx** [comment]
> It's worth noting for the servers this affects, if worldedit doesn't have the threadlocal the server still dies from Brigadier's threadlocal

Yes but we haven't seen Brigadier crashed a server.

Maybe just because player interaction events are way more frequent.

File: worldedit-core/src/main/java/com/sk89q/worldedit/session/request/Request.java [hunk 1]
```diff
@@ -24,13 +24,15 @@
...
+import java.util.Map;
 
 /**
- * Describes the current request using a {@link ThreadLocal}.
+ * Describes the current request using a Thread.name() -> request map(avoid server freeze on scale of 50+ players).
  */
 public final class Request {
 
-    private static final ThreadLocal<Request> threadLocal = ThreadLocal.withInitial(Request::new);
+    private static Map<String, Request> reqMap = new HashMap<>();
```
Comment (zml2008) [inline worldedit-core/src/main/java/com/sk89q/worldedit/session/request/Request.java:35]: HashMap is also not thread-safe

**Review by zml2008 — CHANGES_REQUESTED** [review]

File: worldedit-core/src/main/java/com/sk89q/worldedit/session/request/Request.java (hunk 1 above)
Comment (ReinWD) [reply worldedit-core/src/main/java/com/sk89q/worldedit/session/request/Request.java:35]: So how about changing the key of this HashMap?
like thread id or something

Comment (zml2008) [reply worldedit-core/src/main/java/com/sk89q/worldedit/session/request/Request.java:35]: Do you know what "thread-safe" means?

Comment (ReinWD) [reply worldedit-core/src/main/java/com/sk89q/worldedit/session/request/Request.java:35]: considering the visibility, I think if we can find a key to identify each thread, it's no need to achieve full thread safe in this case.

**Comment by octylFractal** [comment]
As noted by @zml2008, this PR introduces a thread-safety bug by using `HashMap`, plus it leaks memory if a request is not removed from the map before a thread closes. I've opened #1862 to fix these issues while still doing a workaround.

**Pull request closed without merging**